tests/test_checksum.py

```diff
@@ -1,34 +1,46 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
-
+import base64
 import hashlib
 import pytest
 
-from cachito.common.checksum import compute_file_checksum, get_hasher
+from cachito.common.checksum import hash_file, Hasher
 from cachito.errors import UnknownHashAlgorithm
 
 
-@pytest.mark.parametrize(
-    "name,expected_error",
-    [
-        ["sha256", None],
-        ["sha512", None],
-        ["md5", None],
-        ["xxx", pytest.raises(UnknownHashAlgorithm)],
-    ],
-)
-def test_get_hasher(name, expected_error):
-    if expected_error is None:
-        hasher = get_hasher(name)
-        assert hasattr(hasher, "hexdigest")
-    else:
-        with expected_error:
-            get_hasher(name)
+def test_get_unknown_hash_algorithm():
+    with pytest.raises(UnknownHashAlgorithm):
+        hash_file("some_file.tar", algorithm="xxx")
 
 
 @pytest.mark.parametrize("file_content", ["", "abc123" * 100])
-def test_compute_file_checksum(file_content, tmpdir):
+@pytest.mark.parametrize("algorithm", [None, "sha512"])
+def test_hash_file(file_content, algorithm, tmpdir):
     data_file = tmpdir.join("file.data")
     data_file.write(file_content)
-    expected_checksum = hashlib.sha256(file_content.encode()).hexdigest()
-    hasher = get_hasher("sha256")
-    assert expected_checksum == compute_file_checksum(hasher, str(data_file))
+
+    if algorithm is None:
+        hasher = hash_file(str(data_file))
+        assert hashlib.sha256(file_content.encode()).digest() == hasher.digest()
+    else:
+        hasher = hash_file(str(data_file), algorithm=algorithm)
+        h = hashlib.new(algorithm)
+        h.update(file_content.encode())
+        assert h.digest() == hasher.digest()
+
+
+def test_hasher_digest():
+    h = hashlib.new("sha256")
+    hasher = Hasher(h)
+    assert h.digest() == hasher.digest()
+
+
+def test_hasher_hexdigest():
+    h = hashlib.new("sha256")
+    hasher = Hasher(h)
+    assert h.hexdigest() == hasher.hexdigest()
+
+
+def test_hasher_base64digest():
+    h = hashlib.new("sha256")
+    hasher = Hasher(h)
+    assert base64.b64encode(h.digest()).decode("utf-8") == hasher.base64digest()
```

Comment:
I'm not a big fan of tests that are essentially just a copy-paste of the implementation, maybe the expected value could be hardcoded instead?